controller/posts-controller.js

```diff
@@ -55,15 +55,14 @@ exports.getSinglePost = async (req, res) => {
     return;
   }
 
-  let post = Posts.findById(postId).populate("userid", "userName");
+  let post = await Post.findById(postId).populate("user", "userName");
```

Comment:
Again, why you're using `let` here?

Reply:
I am using `post` again in code..

Reply:
You can combine the next populate expression to this expression itself.

Reply:
updated